components/elm/src/biogeochem/EcosystemDynMod.F90

```diff
@@ -575,382 +517,350 @@ subroutine EcosystemDynNoLeaching2(bounds,                                  &
-            call t_stopf('phenology_flux_limiter')
+          event = 'phenology_flux_limiter'
+          call t_startGPU(event)
+          call phenology_flux_limiter(bounds, num_soilc, filter_soilc,&
+            num_soilp, filter_soilp, crop_vars, cnstate_vars,  &
+            veg_cf, veg_cs, &
+            c13_veg_cf, c13_veg_cs, &
+            c14_veg_cf, c14_veg_cs, &
+            veg_nf, veg_ns, veg_pf, veg_ps)
+          call t_stopGPU(event)
         endif
-        call t_startf('CNLitterToColumn')
-        call CNLitterToColumn(num_soilc, filter_soilc, &
-              cnstate_vars, carbonflux_vars, nitrogenflux_vars,phosphorusflux_vars)
+        event = 'CNLitterToColumn'
+        call t_startGPU(event)
+        call CNLitterToColumn(num_soilp, filter_soilp, cnstate_vars )
+        call t_stopGPU(event)
 
-        call t_stopf('CNLitterToColumn') 
-        
         !--------------------------------------------
         ! Update1
         !--------------------------------------------
-        
-        call t_startf('CNUpdate1')
-        
+        event = 'CNUpdate1'
+        call t_startGPU(event)
+
         if ( use_c13 ) then
-            call CarbonIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
-                  cnstate_vars, carbonflux_vars, carbonstate_vars, &
-                  isotopeflux_vars=c13_carbonflux_vars, isotopestate_vars=c13_carbonstate_vars, &
-                  isotope='c13', isocol_cs=c13_col_cs, isoveg_cs=c13_veg_cs, isocol_cf=c13_col_cf, isoveg_cf=c13_veg_cf)
+          call CarbonIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
+               cnstate_vars, &
+               isotope=c13, isocol_cs=c13_col_cs, isoveg_cs=c13_veg_cs, isocol_cf=c13_col_cf, isoveg_cf=c13_veg_cf)
         end if
-        
+
         if ( use_c14 ) then
-            call CarbonIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
-                  cnstate_vars, carbonflux_vars, carbonstate_vars, &
-                  isotopeflux_vars=c14_carbonflux_vars, isotopestate_vars=c14_carbonstate_vars, &
-                  isotope='c14', isocol_cs=c14_col_cs, isoveg_cs=c14_veg_cs, isocol_cf=c14_col_cf, isoveg_cf=c14_veg_cf)
+          call CarbonIsoFlux1(num_soilc, filter_soilc, num_soilp, filter_soilp, &
+               cnstate_vars , &
+               isotope=c14, isocol_cs=c14_col_cs, isoveg_cs=c14_veg_cs, isocol_cf=c14_col_cf, isoveg_cf=c14_veg_cf)
          end if
-         
+
     end if  ! if(.not.use_fates)
 
     if(use_fates) then
-       ! In this scenario, we simply zero all of the 
+       ! In this scenario, we simply zero all of the
        ! column level variables that would had been upscaled
        ! in the veg summary with p2c
        call col_cf%ZeroForFatesRR(bounds,num_soilc, filter_soilc)
 
        ! Transfer fates litter fluxes into ELM source arrays
-       call elm_fates%UpdateLitterFluxes(bounds)
+       call alm_fates%UpdateLitterFluxes(bounds)
```

Comment:
please change it back to `elm_fates`

cc: @rgknox

Reply:
There's more instances of changing alm_fates from being passed an argument to using it directly through elm_instMod.  This way #ifndef _OPENACC flags can be used around these subroutine calls and the alm_fates data type doesn't need a device copy.